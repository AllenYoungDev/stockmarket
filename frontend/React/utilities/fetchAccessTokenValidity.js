'use strict';

import '../settings/ServerUrls.js';

export function fetchAccessTokenValidity(setLoginState, ignore) {

    console.log('fetchAccessTokenValidity() start.')

    fetch(window.backendServerUrl + "/CheckAccessTokenValidity", {method: "GET", credentials: 'include'})
    .then((response) => response.text())
    .then((text) => {
        if (text === 'valid') {
            console.log('fetchAccessTokenValidity():  access token is valid.');
            if (ignore) {return;}
            if (document.cookie.includes('admin=true')) {
                setLoginState('admin-login');
            } else {
                setLoginState('non-admin-login');
            }
        } else {
            console.log('fetchAccessTokenValidity():  access token is invalid.');
            if (ignore) {return;}
            setLoginState('no-login');
        }
    })
    .catch((err) => {
        console.log(`fetchAccessTokenValidity():  fetch() network error.  ${err}`);
        if (ignore) {return;}
        setLoginState('fetch-error');
    });
}