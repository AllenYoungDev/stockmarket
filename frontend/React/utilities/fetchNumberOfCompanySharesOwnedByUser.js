'use strict';

import '../settings/ServerUrls.js';

export function fetchNumberOfCompanySharesOwnedByUser(setResponseStatusBody, ignore) {
    let responseStatus = 0;

    console.log('fetchNumberOfCompanySharesOwnedByUser() start.')

    fetch(window.backendServerUrl + "/GetNumberOfCompanySharesOwnedByUser", {method: "GET", credentials: 'include'})
    .then((response) => {responseStatus=response.status; return response.text();})
    .then((text) => {
      console.log(`fetchNumberOfCompanySharesOwnedByUser() responseStatus:  ${responseStatus}.`)
      console.log(`fetchNumberOfCompanySharesOwnedByUser() responseBody:  ${text}.`)
      if (!ignore) {
        setResponseStatusBody({responseStatus: responseStatus, responseBody: text});
      }
    })
    .catch((err) => {
      if (!ignore) {
        setResponseStatusBody({responseStatus: -1, responseBody: ''}); //'fetch-error'
      }
    });
}