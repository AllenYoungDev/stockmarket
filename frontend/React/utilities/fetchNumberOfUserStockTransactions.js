import '../settings/ServerUrls.js';

export function fetchNumberOfUserStockTransactions(setNumberOfUserStockTransactions, ignore) {
    let responseStatus = 0;

    fetch(window.backendServerUrl + `/GetNumberOfUserStockTransactions`, {method: "GET", credentials: 'include'})
    .then((response) => {responseStatus = response.status; return response.text();})
    .then((text) => {
      console.log(`CompanyStockTransactionHistory():  /GetNumberOfUserStockTransactions fetch() response status:  ${responseStatus}.`);
      console.log(`CompanyStockTransactionHistory():  /GetNumberOfUserStockTransactions fetch() response:  ${text}.`);

      if (ignore) {return;}

      if( responseStatus === 200) {
        setNumberOfUserStockTransactions(parseInt(text, 10));
      } else {
        setNumberOfUserStockTransactions(0);
      }
    })
    .catch((err) => {
      console.log(`CompanyStockTransactionHistory():  /GetNumberOfUserStockTransactions fetch() network error.  ${err}`);
      if (ignore) {return;}
      setNumberOfUserStockTransactions(0);
    });
}