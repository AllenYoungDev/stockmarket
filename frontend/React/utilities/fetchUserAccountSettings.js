import '../settings/ServerUrls.js';

export function fetchUserAccountSettings(setUserAccountSettings, ignore) {
    let responseStatus = 0;
    
    fetch(window.backendServerUrl + "/GetUserAccountSettings", {method: "GET", credentials: 'include'})
    .then((response) => {
        responseStatus = response.status
        console.log(`useUserAccountSettings() responseStatus:  ${responseStatus}.`)

        if (responseStatus === 200) {
            return response.json();
        } else {
            return response.text();
        }
    })
    .then((data) => {
        if (ignore) {return;}

        console.log(`useUserAccountSettings() data:  ${JSON.stringify(data)}.`)

        if (responseStatus === 200) {
            setUserAccountSettings(data);
        } else {
            setUserAccountSettings(null);
        }
    })
    .catch((err) => {
      console.log(`useUserAccountSettings():  fetch() network error.  ${err}`);
      if (ignore) {return;}
      setUserAccountSettings(null);
    });
}