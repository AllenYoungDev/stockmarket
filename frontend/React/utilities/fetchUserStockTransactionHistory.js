import '../settings/ServerUrls.js';

export function fetchUserStockTransactionHistory(pageNumber, setUserStockTransactionHistory, ignore) {
  let responseStatus = 0;

  fetch(window.backendServerUrl + `/GetUserStockTransactionHistory/${pageNumber}`, {method: "GET", credentials: 'include'})
  .then((response) => {
    responseStatus = response.status

    if (responseStatus === 200) {
        return response.json();
    } else {
        return response.text();
    }
  })
  .then((data) => {
    if (ignore) {return;}

    console.log(`CompanyStockTransactionHistory() /GetUserStockTransactionHistory/:pageNumber fetch() data: ${JSON.stringify(data)}.`)

    if( responseStatus === 200) {
      setUserStockTransactionHistory(data);
    } else {
      setUserStockTransactionHistory([]);
    }
  })
  .catch((err) => {
    console.log(`CompanyStockTransactionHistory():  /GetUserStockTransactionHistory/:pageNumber fetch() network error.  ${err}`);
    if (ignore) {return;}
    setUserStockTransactionHistory([]);
  });
}