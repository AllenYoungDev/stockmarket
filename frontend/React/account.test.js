/**
 * @jest-environment jsdom
 */

//import React from "react";
//import ReactDOM from 'react-dom/client';

import renderer from 'react-test-renderer';
import {UserAccountPage, AdminLogout, AccountSettingsUpdateForm, CompanyStockTransactionHistory} from './account.js';

jest.mock('./utilities/fetchAccessTokenValidity');
jest.mock('./utilities/fetchNumberOfCompanySharesOwnedByUser');
jest.mock('./utilities/fetchNumberOfUserStockTransactions');
jest.mock('./utilities/fetchUserStockTransactionHistory');
jest.mock('./utilities/fetchUserAccountSettings');

import {fetchAccessTokenValidity} from './utilities/fetchAccessTokenValidity.js';
import {fetchNumberOfCompanySharesOwnedByUser} from './utilities/fetchNumberOfCompanySharesOwnedByUser.js';
import {fetchNumberOfUserStockTransactions} from './utilities/fetchNumberOfUserStockTransactions.js';
import {fetchUserStockTransactionHistory} from './utilities/fetchUserStockTransactionHistory.js';
import {fetchUserAccountSettings} from './utilities/fetchUserAccountSettings.js';

(async () => {
  if (typeof React === 'undefined') {
    globalThis.React = await import("react");
  }
  if (typeof ReactDOM === 'undefined') {
    globalThis.ReactDOM = await import("react-dom/client");
  }
})();

var loginState = '';

async function userAccountPageJestSnapshotTest(loginStateArg) {
  let testRenderer;
  let testInstance;
  let tree;

  loginState = loginStateArg;

  //renderer.create() is wrapped in renderer.act() for executing useEffect() calls in <HomePage />.
  //https://github.com/facebook/react/issues/15321
  await renderer.act(async () => {
    testRenderer = renderer.create(
      <UserAccountPage />,
    );
  });

  testInstance = testRenderer.root;
  expect(testInstance.findByType(AdminLogout).props.loginState).toBe(loginStateArg);
  expect(testInstance.findByType(AccountSettingsUpdateForm).props.loginState).toBe(loginStateArg);
  expect(testInstance.findByType(CompanyStockTransactionHistory).props.loginState).toBe(loginStateArg);

  tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
}

it('React app Jest snapshot testing for Allen Young\'s Stockmarket Demo user account page', async () => {
  const numberOfCompanySharesOwnedByUserSuccessResponse = {responseStatus: 200, responseBody: '1121'};
  const numberOfCompanySharesOwnedByUserFetchError = {responseStatus: -1, responseBody: ''};

  fetchAccessTokenValidity.mockImplementation((setLoginState, ignore) => {
    console.log('fetchAccessTokenValidity() mock called!');
    setLoginState(loginState);
  });
  fetchNumberOfCompanySharesOwnedByUser.mockImplementation((setResponseStatusBody, ignore) => {
    console.log('fetchNumberOfCompanySharesOwnedByUser() mock called!');
    setResponseStatusBody(numberOfCompanySharesOwnedByUserSuccessResponse);
  });
  fetchNumberOfUserStockTransactions.mockImplementation((setNumberOfUserStockTransactions, ignore) => {
    console.log('fetchNumberOfUserStockTransactions() mock called!');
    setNumberOfUserStockTransactions(0);
  });
  fetchUserStockTransactionHistory.mockImplementation((pageNumber, setUserStockTransactionHistory, ignore) => {
    console.log('fetchUserStockTransactionHistory() mock called!');
    setUserStockTransactionHistory([]);
  });
  fetchUserAccountSettings.mockImplementation((setUserAccountSettings, ignore) => {
    console.log('fetchUserAccountSettings() mock called!');
    setUserAccountSettings(null);
  });

  /*************************************************************************************************
  No-login mock Jest snapshot test
  **************************************************************************************************/
  await userAccountPageJestSnapshotTest('no-login');

  /*************************************************************************************************
  Non-admin-user-login mock Jest snapshot test
  **************************************************************************************************/
  await userAccountPageJestSnapshotTest('non-admin-login');

  /*************************************************************************************************
  Admin-user-login mock Jest snapshot test
  **************************************************************************************************/
  await userAccountPageJestSnapshotTest('admin-login');

  /*************************************************************************************************
  Fetch-error mock Jest snapshot test
  **************************************************************************************************/
  fetchNumberOfCompanySharesOwnedByUser.mockImplementation((setResponseStatusBody, ignore) => {
    console.log('fetchNumberOfCompanySharesOwnedByUser() mock called!');
    setResponseStatusBody(numberOfCompanySharesOwnedByUserFetchError);
  });

  await userAccountPageJestSnapshotTest('fetch-error');
});