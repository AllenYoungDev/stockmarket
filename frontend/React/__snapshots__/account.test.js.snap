// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo user account page 1`] = `
[
  <div
    className="top-bar"
  >
    <div
      className="top-bar-left-column"
    >
      <a
        href="http://localhost"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Allen Young's Stockmarket Demo
      </a>
    </div>
    <div
      className="top-bar-right-column"
    >
      <a
        href="http://localhost/registration.html"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Register
      </a>
       | 
      <a
        href="http://localhost/login.html"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Login
      </a>
    </div>
  </div>,
  <section
    className="account-settings-section"
  >
    <h1
      className="account-settings-section-heading"
    >
      Account Settings
    </h1>
    <form
      className="account-settings-update-form"
    >
      <div
        className="account-settings-update-form-div-grid"
      >
        <label
          className="account-settings-update-form-label-first-name"
          htmlFor="fname"
        >
          First name
        </label>
        <input
          className="account-settings-update-form-input-first-name"
          disabled={true}
          id="fname"
          name="fname"
          size="20"
          type="text"
        />
        <label
          className="account-settings-update-form-label-last-name"
          htmlFor="lname"
        >
          Last name
        </label>
        <input
          className="account-settings-update-form-input-last-name"
          disabled={true}
          id="lname"
          name="lname"
          size="20"
          type="text"
        />
        <div
          className="account-settings-update-form-div-below-last-name"
        />
        <label
          className="account-settings-update-form-label-email-address"
          htmlFor="emailaddress"
        >
          Email address
        </label>
        <input
          className="account-settings-update-form-input-email-address"
          disabled={true}
          id="emailaddress"
          name="emailaddress"
          size="20"
          type="text"
        />
        <label
          className="account-settings-update-form-label-phone-number"
          htmlFor="phonenumber"
        >
          Phone number
        </label>
        <input
          className="account-settings-update-form-input-phone-number"
          disabled={true}
          id="phonenumber"
          name="phonenumber"
          size="20"
          type="text"
        />
        <div
          className="account-settings-update-form-div-below-phone-number"
        />
        <label
          className="account-settings-update-form-label-password"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="account-settings-update-form-input-password"
          disabled={true}
          id="password"
          name="password"
          size="20"
          type="text"
        />
        <label
          className="account-settings-update-form-label-password-retype"
          htmlFor="passwordretype"
        >
          Password (Retype)
        </label>
        <input
          className="account-settings-update-form-input-password-retype"
          disabled={true}
          id="passwordretype"
          name="passwordretype"
          size="20"
          type="text"
        />
      </div>
      <p
        className="account-settings-update-form-notice1"
      >
        Password must be at least 10 characters, and must contain a letter, a number, and a special character from any of the following ~\`!@#$%^&*-_+=|\\/?;:"'&lt;&gt;,.
        []().
      </p>
      <p
        className="account-settings-update-form-notice2"
      >
        Your email address is your login ID and PayPal account ID.
      </p>
      <p
        className="account-settings-update-form-notice3"
      >
        Your email address will be updated instantly without the email-address verification.  Please make sure that the new email address you enter is correct.
      </p>
      <div
        className="account-settings-update-form-div-button"
      >
        <p
          className="account-settings-update-form-error-notice"
          role="error-notice"
        />
        <input
          className="account-settings-update-form-input-button"
          disabled={true}
          type="button"
          value="Update"
        />
      </div>
    </form>
    <div>
      <p
        className="number-of-company-shares-owned"
      >
        Number of Company shares owned by me:  
        1121
      </p>
    </div>
  </section>,
  <section
    className="company-stock-transaction-history-section"
  >
    <h1
      className="company-stock-transaction-history-section-heading"
    >
      Company Stock Transaction History
    </h1>
    <div
      className="company-stock-transaction-history-div-display"
    >
      No records to show.
    </div>
    <p
      className="company-stock-transaction-history-p-navigation"
    >
      &lt; 1 2 3 … (last page number) &gt;
    </p>
    <form
      className="company-stock-transaction-history-form"
    >
      <label
        htmlFor="page_number"
      >
        Page
      </label>
      <input
        disabled={true}
        id="page_number"
        name="page_number"
        size="5"
        type="text"
      />
      <input
        disabled={true}
        type="button"
        value="Go to"
      />
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      © 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo user account page 2`] = `
[
  <div
    className="top-bar"
  >
    <div
      className="top-bar-left-column"
    >
      <a
        href="http://localhost"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Allen Young's Stockmarket Demo
      </a>
    </div>
    <div
      className="top-bar-right-column"
    >
      <a
        href=""
        onClick={[Function]}
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Logout
      </a>
    </div>
  </div>,
  <section
    className="account-settings-section"
  >
    <h1
      className="account-settings-section-heading"
    >
      Account Settings
    </h1>
    <form
      className="account-settings-update-form"
    >
      <div
        className="account-settings-update-form-div-grid"
      >
        <label
          className="account-settings-update-form-label-first-name"
          htmlFor="fname"
        >
          First name
        </label>
        <input
          className="account-settings-update-form-input-first-name"
          id="fname"
          name="fname"
          placeholder="First name"
          size="20"
          type="text"
        />
        <label
          className="account-settings-update-form-label-last-name"
          htmlFor="lname"
        >
          Last name
        </label>
        <input
          className="account-settings-update-form-input-last-name"
          id="lname"
          name="lname"
          placeholder="Last name"
          size="20"
          type="text"
        />
        <div
          className="account-settings-update-form-div-below-last-name"
        />
        <label
          className="account-settings-update-form-label-email-address"
          htmlFor="emailaddress"
        >
          Email address
        </label>
        <input
          className="account-settings-update-form-input-email-address"
          id="emailaddress"
          name="emailaddress"
          placeholder="Email address"
          size="20"
          type="text"
        />
        <label
          className="account-settings-update-form-label-phone-number"
          htmlFor="phonenumber"
        >
          Phone number
        </label>
        <input
          className="account-settings-update-form-input-phone-number"
          id="phonenumber"
          name="phonenumber"
          placeholder="Phone number"
          size="20"
          type="text"
        />
        <div
          className="account-settings-update-form-div-below-phone-number"
        />
        <label
          className="account-settings-update-form-label-password"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="account-settings-update-form-input-password"
          id="password"
          name="password"
          placeholder="Password"
          size="20"
          type="password"
        />
        <p
          className="account-settings-update-form-password-show-hide"
        >
          <a
            href=""
            onClick={[Function]}
            role="password-show-hide"
            style={
              {
                "color": "rgb(255,255,255)",
                "textDecoration": "none",
              }
            }
          >
            show
          </a>
        </p>
        <label
          className="account-settings-update-form-label-password-retype"
          htmlFor="passwordretype"
        >
          Password (Retype)
        </label>
        <input
          className="account-settings-update-form-input-password-retype"
          id="passwordretype"
          name="passwordretype"
          placeholder="Password (Retype)"
          size="20"
          type="password"
        />
      </div>
      <p
        className="account-settings-update-form-notice1"
      >
        Password must be at least 10 characters, and must contain a letter, a number, and a special character from any of the following ~\`!@#$%^&*-_+=|\\/?;:"'&lt;&gt;,.
        []().
      </p>
      <p
        className="account-settings-update-form-notice2"
      >
        Your email address is your login ID and PayPal account ID.
      </p>
      <p
        className="account-settings-update-form-notice3"
      >
        Your email address will be updated instantly without the email-address verification.  Please make sure that the new email address you enter is correct.
      </p>
      <div
        className="account-settings-update-form-div-button"
      >
        <p
          className="account-settings-update-form-error-notice"
          role="error-notice"
        />
        <input
          className="account-settings-update-form-input-button"
          onClick={[Function]}
          type="button"
          value="Update"
        />
      </div>
    </form>
    <div>
      <p
        className="number-of-company-shares-owned"
      >
        Number of Company shares owned by me:  
        1121
      </p>
    </div>
  </section>,
  <section
    className="company-stock-transaction-history-section"
  >
    <h1
      className="company-stock-transaction-history-section-heading"
    >
      Company Stock Transaction History
    </h1>
    <div
      className="company-stock-transaction-history-div-display"
    >
      <table
        className="company-stock-transaction-history-table"
      >
        <thead>
          <tr>
            <th>
              Transaction start datetime
            </th>
            <th>
              PayPal transaction (order) ID
            </th>
            <th>
              Number of shares
            </th>
            <th>
              Payment processing initiated
            </th>
            <th>
              Payment processing completed
            </th>
            <th>
              Payment processing status
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <p
      className="company-stock-transaction-history-p-navigation"
    >
      <a
        href=""
        onClick={[Function]}
        style={
          {
            "color": "#36454F",
            "textDecoration": "none",
          }
        }
      >
        &lt;
      </a>
      <span>
         
      </span>
      <a
        href=""
        onClick={[Function]}
        style={
          {
            "color": "#36454F",
            "textDecoration": "none",
          }
        }
      >
        &gt;
      </a>
    </p>
    <form
      className="company-stock-transaction-history-form"
    >
      <label
        htmlFor="page_number"
      >
        Page
      </label>
      <input
        id="page_number"
        max={0}
        min="1"
        name="page_number"
        onInput={[Function]}
        size="5"
        step="1"
        type="number"
      />
      <input
        onClick={[Function]}
        type="button"
        value="Go to"
      />
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      © 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo user account page 3`] = `
[
  <div
    className="top-bar"
  >
    <div
      className="top-bar-left-column"
    >
      <a
        href="http://localhost"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Allen Young's Stockmarket Demo
      </a>
    </div>
    <div
      className="top-bar-right-column"
    >
      <a
        href="http://localhost/admin.html"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Admin
      </a>
       | 
      <a
        href=""
        onClick={[Function]}
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Logout
      </a>
    </div>
  </div>,
  <section
    className="account-settings-section"
  >
    <h1
      className="account-settings-section-heading"
    >
      Account Settings
    </h1>
    <form
      className="account-settings-update-form"
    >
      <div
        className="account-settings-update-form-div-grid"
      >
        <label
          className="account-settings-update-form-label-first-name"
          htmlFor="fname"
        >
          First name
        </label>
        <input
          className="account-settings-update-form-input-first-name"
          id="fname"
          name="fname"
          placeholder="First name"
          size="20"
          type="text"
        />
        <label
          className="account-settings-update-form-label-last-name"
          htmlFor="lname"
        >
          Last name
        </label>
        <input
          className="account-settings-update-form-input-last-name"
          id="lname"
          name="lname"
          placeholder="Last name"
          size="20"
          type="text"
        />
        <div
          className="account-settings-update-form-div-below-last-name"
        />
        <label
          className="account-settings-update-form-label-email-address"
          htmlFor="emailaddress"
        >
          Email address
        </label>
        <input
          className="account-settings-update-form-input-email-address"
          id="emailaddress"
          name="emailaddress"
          placeholder="Email address"
          size="20"
          type="text"
        />
        <label
          className="account-settings-update-form-label-phone-number"
          htmlFor="phonenumber"
        >
          Phone number
        </label>
        <input
          className="account-settings-update-form-input-phone-number"
          id="phonenumber"
          name="phonenumber"
          placeholder="Phone number"
          size="20"
          type="text"
        />
        <div
          className="account-settings-update-form-div-below-phone-number"
        />
        <label
          className="account-settings-update-form-label-password"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="account-settings-update-form-input-password"
          id="password"
          name="password"
          placeholder="Password"
          size="20"
          type="password"
        />
        <p
          className="account-settings-update-form-password-show-hide"
        >
          <a
            href=""
            onClick={[Function]}
            role="password-show-hide"
            style={
              {
                "color": "rgb(255,255,255)",
                "textDecoration": "none",
              }
            }
          >
            show
          </a>
        </p>
        <label
          className="account-settings-update-form-label-password-retype"
          htmlFor="passwordretype"
        >
          Password (Retype)
        </label>
        <input
          className="account-settings-update-form-input-password-retype"
          id="passwordretype"
          name="passwordretype"
          placeholder="Password (Retype)"
          size="20"
          type="password"
        />
      </div>
      <p
        className="account-settings-update-form-notice1"
      >
        Password must be at least 10 characters, and must contain a letter, a number, and a special character from any of the following ~\`!@#$%^&*-_+=|\\/?;:"'&lt;&gt;,.
        []().
      </p>
      <p
        className="account-settings-update-form-notice2"
      >
        Your email address is your login ID and PayPal account ID.
      </p>
      <p
        className="account-settings-update-form-notice3"
      >
        Your email address will be updated instantly without the email-address verification.  Please make sure that the new email address you enter is correct.
      </p>
      <div
        className="account-settings-update-form-div-button"
      >
        <p
          className="account-settings-update-form-error-notice"
          role="error-notice"
        />
        <input
          className="account-settings-update-form-input-button"
          onClick={[Function]}
          type="button"
          value="Update"
        />
      </div>
    </form>
    <div>
      <p
        className="number-of-company-shares-owned"
      >
        Number of Company shares owned by me:  
        1121
      </p>
    </div>
  </section>,
  <section
    className="company-stock-transaction-history-section"
  >
    <h1
      className="company-stock-transaction-history-section-heading"
    >
      Company Stock Transaction History
    </h1>
    <div
      className="company-stock-transaction-history-div-display"
    >
      No records to show.
    </div>
    <p
      className="company-stock-transaction-history-p-navigation"
    >
      &lt; 1 2 3 … (last page number) &gt;
    </p>
    <form
      className="company-stock-transaction-history-form"
    >
      <label
        htmlFor="page_number"
      >
        Page
      </label>
      <input
        disabled={true}
        id="page_number"
        name="page_number"
        size="5"
        type="text"
      />
      <input
        disabled={true}
        type="button"
        value="Go to"
      />
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      © 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo user account page 4`] = `
[
  <div
    className="top-bar"
  >
    <div
      className="top-bar-left-column"
    >
      <a
        href="http://localhost"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Allen Young's Stockmarket Demo
      </a>
    </div>
    <div
      className="top-bar-right-column"
    >
      <div>
        Network error.  Please retry later.
      </div>
    </div>
  </div>,
  <section
    className="account-settings-section"
  >
    <h1
      className="account-settings-section-heading"
    >
      Account Settings
    </h1>
    <form
      className="account-settings-update-form"
    >
      <div
        className="account-settings-update-form-div-grid"
      >
        <label
          className="account-settings-update-form-label-first-name"
          htmlFor="fname"
        >
          First name
        </label>
        <input
          className="account-settings-update-form-input-first-name"
          disabled={true}
          id="fname"
          name="fname"
          size="20"
          type="text"
        />
        <label
          className="account-settings-update-form-label-last-name"
          htmlFor="lname"
        >
          Last name
        </label>
        <input
          className="account-settings-update-form-input-last-name"
          disabled={true}
          id="lname"
          name="lname"
          size="20"
          type="text"
        />
        <div
          className="account-settings-update-form-div-below-last-name"
        />
        <label
          className="account-settings-update-form-label-email-address"
          htmlFor="emailaddress"
        >
          Email address
        </label>
        <input
          className="account-settings-update-form-input-email-address"
          disabled={true}
          id="emailaddress"
          name="emailaddress"
          size="20"
          type="text"
        />
        <label
          className="account-settings-update-form-label-phone-number"
          htmlFor="phonenumber"
        >
          Phone number
        </label>
        <input
          className="account-settings-update-form-input-phone-number"
          disabled={true}
          id="phonenumber"
          name="phonenumber"
          size="20"
          type="text"
        />
        <div
          className="account-settings-update-form-div-below-phone-number"
        />
        <label
          className="account-settings-update-form-label-password"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="account-settings-update-form-input-password"
          disabled={true}
          id="password"
          name="password"
          size="20"
          type="text"
        />
        <label
          className="account-settings-update-form-label-password-retype"
          htmlFor="passwordretype"
        >
          Password (Retype)
        </label>
        <input
          className="account-settings-update-form-input-password-retype"
          disabled={true}
          id="passwordretype"
          name="passwordretype"
          size="20"
          type="text"
        />
      </div>
      <p
        className="account-settings-update-form-notice1"
      >
        Password must be at least 10 characters, and must contain a letter, a number, and a special character from any of the following ~\`!@#$%^&*-_+=|\\/?;:"'&lt;&gt;,.
        []().
      </p>
      <p
        className="account-settings-update-form-notice2"
      >
        Your email address is your login ID and PayPal account ID.
      </p>
      <p
        className="account-settings-update-form-notice3"
      >
        Your email address will be updated instantly without the email-address verification.  Please make sure that the new email address you enter is correct.
      </p>
      <div
        className="account-settings-update-form-div-button"
      >
        <p
          className="account-settings-update-form-error-notice"
          role="error-notice"
        />
        <input
          className="account-settings-update-form-input-button"
          disabled={true}
          type="button"
          value="Update"
        />
      </div>
    </form>
    <p
      className="number-of-company-shares-owned"
    >
      Network error.  Please retry later.
    </p>
  </section>,
  <section
    className="company-stock-transaction-history-section"
  >
    <h1
      className="company-stock-transaction-history-section-heading"
    >
      Company Stock Transaction History
    </h1>
    <div
      className="company-stock-transaction-history-div-display"
    >
      No records to show.
    </div>
    <p
      className="company-stock-transaction-history-p-navigation"
    >
      &lt; 1 2 3 … (last page number) &gt;
    </p>
    <form
      className="company-stock-transaction-history-form"
    >
      <label
        htmlFor="page_number"
      >
        Page
      </label>
      <input
        disabled={true}
        id="page_number"
        name="page_number"
        size="5"
        type="text"
      />
      <input
        disabled={true}
        type="button"
        value="Go to"
      />
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      © 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;
